lib/libc/stdio/lib_lowoutstream.c

```diff
@@ -76,14 +76,11 @@
 static void lowoutstream_putc(FAR struct lib_outstream_s *this, int ch)
 {
 	DEBUGASSERT(this);
-#if defined(CONFIG_BUILD_PROTECTED)
-	if (putchar(ch) != EOF)
```

Comment:
This change was taken by mistake from a previous PR 
I have reverted it.

Reply:
I have reverted this to original version.
lowoutputstream API is used for lowsysdebug or lldbg scenario. It is used only in kernel. 
If this change is not taken, then boot up gets stalled since lldbg will not work properly.

Reply:
I agree with this change as there is no requirement of lldbg from userspace. 
However i feel, in broader scope there is no requirement of lowsyslog() from userspace. 
Is it possible to restrict lowsyslog/lldbg() only to `__KERNEL__` incase of `PROTECTED?`
@jeongarmy  would help more on this

Userspace : only API's which use posix streams
Kernel : Lets keep all variants